pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -323,6 +323,20 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 				// TODO: Skip dynamic provisioning for now
 				ctrl.eventRecorder.Event(claim, v1.EventTypeNormal, events.WaitForFirstConsumer, "waiting for first consumer to be created before binding")
 			case v1helper.GetPersistentVolumeClaimClass(claim) != "":
+				volume, err := ctrl.volumes.findDynamicProvisionedVolumeForClaim(claim)
```

Comment:
Wouldn't a volume have been returned in `findBestMatchForClaim`?

Reply:
As findBestMatchForClaim calls checkVolumeModeMisMatches inside, the volume returned by this function excludes the volume which has volumeMode mismatch(*).
Therefore, in volumeMode mismatch case, no volume is returned by findBestMatchForClaim.

(*) findBestMatchForClaim()->findByClaim()->findMatchingVolume()->checkVolumeModeMisMatches()